src/devices/CharacterLcd/Hd44780.cs

```diff
@@ -16,46 +17,54 @@ namespace Iot.Device.CharacterLcd
+    /// Samsung KS0066U, Aiptek AIP31066, and many more.
     /// 
     /// Some compatible chips extend the HD44780 with addtional pins and features. They are still fully compatible. The ST7036 is one example.
     /// 
     /// This implementation was drawn from numerous datasheets and libraries such as Adafruit_Python_CharLCD.
     /// </remarks>
-    public class Hd44780 : Hd44780Base, IDisposable
+    public class Hd44780 : IDisposable
     {
-        /// <summary>
-        /// Register select pin. Low is for writing to the instruction
-        /// register and reading the address counter. High is for reading
-        /// and writing to the data register.
-        /// </summary>
-        private readonly int _rsPin;
-
-        /// <summary>
-        /// Read/write pin. Low for write, high for read.
-        /// </summary>
-        private readonly int _rwPin;
+        private bool _disposed;
 
-        /// <summary>
-        /// Enable pin. Pulse high to process a read/write.
-        /// </summary>
-        private readonly int _enablePin;
+        protected const byte ClearDisplayCommand = 0b_0001;
+        protected const byte ReturnHomeCommand = 0b_0010;
 
-        private readonly int _backlight;
+        protected const byte SetCGRamAddressCommand = 0b_0100_0000;
+        protected const byte SetDDRamAddressCommand = 0b_1000_0000;
 
-        private readonly int[] _dataPins;
+        internal DisplayFunction _displayFunction = DisplayFunction.Command;
+        internal DisplayControl _displayControl = DisplayControl.Command;
+        internal DisplayEntryMode _displayMode = DisplayEntryMode.Command;
 
-        private IGpioController _controller;
+        protected readonly byte[] _rowOffsets;
 
-        private byte _lastByte;
-        private bool _useLastByte;
+        protected readonly LcdInterface _interface;
 
-        private readonly PinValuePair[] _pinBuffer = new PinValuePair[8];
+        /// <summary>
+        /// The command wait time multiplier for the LCD.
+        /// </summary>
+        /// <remarks>
+        /// Timings are based off the original HD44780 specification, which dates back
+        /// to the late 1980s. Modern LCDs can often respond faster. In addition we spend
+        /// time in other code between pulsing in new data, which gives quite a bit of
+        /// headroom over the by-the-book waits.
+        /// 
+        /// This is more useful if you have a slow GpioAdapter as time spent in the
+        /// adapter may eat into the need to wait as long for a command to complete.
+        /// 
+        /// There is a busy signal that can be checked that could make this moot, but
+        /// currently we are unable to check the signal fast enough to make gains (or
+        /// even equal) going off hard timings. The busy signal also requires having a
+        /// r/w pin attached. And lastly, as already stated, it takes time to set up
+        /// for another byte of data to be pulsed in.
```

Comment:
Is this expected to be set to value between 0..1 where 1.0 is the slowest possible display (the most compat) and some much lower value for faster displays or reverse (1+ and current value optimizes for most of the cases)? I think the summary should clarify that since this seems a bit confusing property.

Reply:
I changed the name to `WaitMultiplier` and moved to LcdInterface (it got disconnected).